javascript/elements/updates_for_element.js

```diff
@@ -45,45 +45,58 @@ export default class UpdatesForElement extends SubscribingElement {
     }
   }
 
-  shouldUpdate (data) {
-    return (
-      !this.ignoringInnerUpdates &&
-      this.hasChangesSelectedForUpdate(data) &&
-      this.blocks[0] === this
-    )
-  }
-
   update (data) {
-    ActiveElement.set(document.activeElement)
+    // memoize blocks to avoid unnecessary DOM traversal
+    this.blocks = document.querySelectorAll(this.query)
```

Comment:
I'd still move this to a `get`ter for better structure, but memoized, i.e.

```js
get blocks() {
  if(!this._blocks) this._blocks = document.querySelectorAll(this.query)
  
  return this._blocks
}
```

Reply:
My hope is to hand this off again, soon - and if you want it to be a getter, it shall be a getter! I localized it for the development because it's only called in twice, and in the same method, too.

Reply:
I just love short methods 😅